### ISSUE TITLE
Size mismatch using checkpoints saved during training (before completion) when they are used for further training with altered number of classes.

### ISSUE THREAD

**Issue by tserdar** [ultralytics/yolov5#6172]
### Search before asking

- [X] I have searched the YOLOv5 [issues](https://github.com/ultralytics/yolov5/issues) and found no similar bug report.

### Bug

I currently have a model being trained for 4 days and it is still running. During which, I save checkpoints every 20 epochs for testing purposes. 

The training process is now at epoch 94/150. I am saving the checkpoints from every 10 epochs. 

If I take any checkpoint saved during this particular training with 2000+ number of classes (i.e. epoch90.pt) and try to use this very file to apply transfer learning for 8 classes, I am getting size mismatch error:

![image](https://user-images.githubusercontent.com/62431982/147948393-5041749a-5391-416a-9754-ec48ab8913be.png)

### Environment

-YOLOv5, release v6.0 branch,
-Ubuntu Focal

### Are you willing to submit a PR?

- [ ] Yes I'd like to help by submitting a PR!

**Comment by glenn-jocher** [comment]
@rocketsfallonrocketfalls 👋 Hello! Thanks for asking about **training checkpoints**.  YOLOv5 🚀 checkpoints should be about 4X the size of final trained checkpoints, as they carry not just a FP16 model, but a FP16 [EMA](https://www.tensorflow.org/api_docs/python/tf/train/ExponentialMovingAverage) and an FP32 optimizer of the same size as the model (each model parameter has its own FP32 gradient saved within the optimizer). Final checkpoints contain only an FP16 model, with the EMA and optimizer both stripped after the final epoch of training:
https://github.com/ultralytics/yolov5/blob/b4a29b5a8d63a8c2d4a8929942b44e8969c5dddd/train.py#L423-L425

The `strip_optimizer()` function updates the checkpoint dictionary value for `model`, replacing it with `ema`, and sets `ema` and `optimizer` keys to `None`, which will reduce the checkpoint size by 3/4.
https://github.com/ultralytics/yolov5/blob/b4a29b5a8d63a8c2d4a8929942b44e8969c5dddd/utils/general.py#L741-L755

Final trained checkpoints after `strip_optimizer()` should match the README table sizes below.
<p align="center"><img width="700" alt="YOLOv5 Models" src="https://github.com/ultralytics/yolov5/releases/download/v1.0/model_comparison.png"></p>

You can also run `strip_optimizer()` manually on any checkpoint to convert it into finalized weights ready to train a new model:
```python
from utils.general import strip_optimizer

strip_optimizer('path/to/best.pt')
!python train.py --weights path/to/best.pt  # use best.pt to train a new model
```

Good luck 🍀 and let us know if you have any other questions!

**Comment by tserdar** [comment]
Solved my issue seamlessly. Thanks!
